experimental/devices/ccs811/ccs811.go

```diff
@@ -0,0 +1,361 @@
+	mesModeValue := (measurementMode << 4)
+	if generateInterrupt {
+		mesModeValue = mesModeValue | (0x1 << 3)
+	}
+	if useThreshold {
+		mesModeValue = mesModeValue | (0x1 << 2)
+	}
+
+	// set measurement mode
+	err := d.c.Tx([]byte{measurementModeReg, mesModeValue}, nil)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+type MeasurementMode struct {
+	MeasurementMode   byte
+	GenerateInterrupt bool
+	UseThreshold      bool
+}
+
+func (d *Dev) GetMeasurementMode() (mm MeasurementMode, err error) {
+	r := make([]byte, 1)
+	err = d.c.Tx([]byte{measurementModeReg}, r)
+	if err != nil {
+		return mm, err
+	}
+	mode := r[0] >> 4
+	threshold := (r[0]&4 == 1)
+	interrupt := (r[0]&8 == 1)
+
+	return MeasurementMode{MeasurementMode: mode, GenerateInterrupt: interrupt, UseThreshold: threshold}, nil
+}
+
+// Reset sets device into the BOOT mode
+func (d *Dev) Reset() error {
+	if err := d.c.Tx([]byte{resetReg, 0x11, 0xE5, 0x72, 0x8A}, nil); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ReadStatus returns value of status register
+func (d *Dev) ReadStatus() (byte, error) {
+	r := make([]byte, 1)
+	if err := d.c.Tx([]byte{statusReg}, r); err != nil {
+		return 0, err
+	}
+	return r[0], nil
+}
+
+// ReadRawData provides current and voltage on the sensor
+// current in uA
+// voltage from 0-1023, where 1023 = 1.65V
+func (d *Dev) ReadRawData() (current int, voltage int, err error) {
+	r := make([]byte, 2)
+	if err = d.c.Tx([]byte{rawDataReg}, r); err != nil {
+		return 0, 0, err
+	}
+	current, voltage = valuesFromRawData(r)
+	return
+}
+
+// SetEnvironmentData allows to provide temperature and humidity so
+// sensor can compensate it's measurement
+func (d *Dev) SetEnvironmentData(temp, humidity float32) error {
+	rawTemp := uint16((temp + 25) * 512)
+	rawHum := uint16(humidity * 512)
+	w := []byte{environmentReg,
+		byte(rawHum >> 8),
+		byte(rawHum),
+		byte(rawTemp >> 8),
+		byte(rawTemp)}
+
+	if err := d.c.Tx(w, nil); err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetBaseline provides current baseline used by measurement alogrithm
+func (d *Dev) GetBaseline() ([]byte, error) {
+	r := make([]byte, 2)
+	if err := d.c.Tx([]byte{baselineReg}, r); err != nil {
+		return r, err
+	}
+	return r, nil
+}
+
+/*
+SetBaseline sets current baseline for measurement algorithm.
+For mor detail check sensor's specification
+
+Manual Baseline Correction
+There is a mechanism within CCS811 to manually save and restore a previously
+saved baseline value using the BASELINE register. The correct time to save the baseline
+will depend on the customer use-case and application.
+
+• For devices which are powered for >24 hours at a time:
+• During the first 500 hours – save the baseline every
+24-48 hours.
+• After the first 500 hours – save the baseline every 5-7 days.
+
+• For devices which are powered <24 hours at a time:
+• If the device is run in, save the baseline before power down
```

Comment:
Please double check how it looks in godoc and tweak. You can run godoc locally with the command `godoc -http=:6060`